ayat/authentication_routes.py

```diff
@@ -0,0 +1,211 @@
+            token = request.headers['x-access-token']
+
+        if not token:
+            return jsonify({'message': 'Token is missing'}), 401
+
+        try:
+            current_user = jwt.decode(token, app.config['SECRET_KEY'])
+
+        except:
+            return jsonify({'message': 'Token is invalid!'}), 401
+        return f(current_user, *args, **kwargs)
+
+    return decorated
+
+
+@app.route('/v1/users', methods=['GET'])
+@token_required
+def get_all_users(current_user):
+    if not current_user['type'] == 'staff':
+        return jsonify({"error": "user is unauthorized"}), 403
+
+    users = User.query.all()
+    output = []
+    for user in users:
+        user_data = {
+            'public_id': user.public_id,
+            'email': user.email,
+            'country_name': user.country_name,
+            'phone_number': user.phone_number,
+            'profile_picture': user.profile_picture,
+            'birth_date': user.birth_date,
+            'gender': user.gender,
+            'registeration_date': user.registeration_date,
+            'is_activated': user.is_activated,
+            'type': user.type
+        }
+
+        output.append(user_data)
+    return jsonify({'users': output}), 200
+
+
+@app.route('/v1/users/<public_id>', methods=['GET'])
+@token_required
+def get_one_user(current_user, public_id):
+
+    if (not current_user['type'] == 'staff') or (not current_user['publid_id'] == str(public_id)):
+        return jsonify({"error": "user is unauthorized"}), 403
+
+    user = User.query.filter_by(public_id=public_id).first()
+    if not user:
+        return jsonify({'message': 'No user found!'})
+
+    user = {
+        'public_id': user.public_id,
+        'email': user.email,
+        'country_name': user.country_name,
+        'phone_number': user.phone_number,
+        'profile_picture': user.profile_picture,
+        'birth_date': user.birth_date,
+        'gender': user.gender,
+        'registeration_date': user.registeration_date,
+        'is_activated': user.is_activated,
+        'type': user.type
+    }
+
+    return jsonify({'user': user}), 200
+
+
+
+@app.route('/v1/users/<public_id>',methods=['PUT'])
+@token_required
+def promote_user(current_user,public_id):
+    if (not current_user['type'] == 'staff') or (not current_user['publid_id'] == str(public_id)):
+        return jsonify({"error": "user is unauthorized"}), 403
+    user = User.query.filter_by(public_id=public_id).first()
+
+    if not user:
+        return jsonify({'message': 'No user found!'})
+
+    data = request.get_json()
+    if current_user['publid_id'] == str(public_id):
+        user.name = data['name']
+        user.email = data['email']
+        user.password = generate_password_hash(data['password'], method='sha256')
+        user.country_name = data['country_name']
+        user.profile_picture = data['profile_picture']
+        user.phone_number = data['phone_number']
+        user.birth_date = data['birth_date']

```

Comment:
We must check if these fields (in the incoming JSON) has a value or even exists, otherwise, we will be erasing all user's data when the request was to just update his avatar. Because all other fields won't exist.